configs/azure-ad.yaml

```diff
@@ -1,6 +1,6 @@
 SOURCE: azure-ad
 TARGET_HOST: graph.microsoft.com
 SOURCE_AUTH_STRATEGY_IDENTIFIER: oauth2_refresh_token
-GRANT_TYPE: client_credentials # by default, assumed to be of type 'urn:ietf:params:oauth:client-assertion-type:jwt-bearer'
+GRANT_TYPE: workload_identity_federation # by default, assumed to be of type 'urn:ietf:params:oauth:client-assertion-type:jwt-bearer'
```

Comment:
Name I've used instead of `client_credentials`; not sure it fits well as it is how its named in GCP and AAD but not in AWS (just federated credentials would fit better?)

Reply:
so this is actually from RFC for OAuth 2.0, and is a value to be passed with the oauth token request:
https://www.rfc-editor.org/rfc/rfc6749#section-4.4.2

Different values depend on what you're requesting an how.  In a sense, the "certificate" was the "client credential", hence this was the type. But it's the most common `grant_type` for most OAuth flows. 

see: https://oauth.net/2/grant-types/

Reply:
i need to review whole PR in more detail, but my thought is that it might make more sense to use a different value for `SOURCE_AUTH_STRATEGY_IDENTIFIER: oauth2_refresh_token` (that is our own invention)

Reply:
after looking through java code, I'm OK with `workload_identity_federation`, although would do some research if there's any convention/standard on this.  Eg, when the grant is authenticated by this trusted-based federation approach - is it still a `client_credential`? or is there a different identifier used? I'd default to what you pass as `grant_type` in the token request, which unless I'm misunderstanding flow, I think is still `client_credentials`, right?

Reply:
It is `client_credentials`, here in MSFT docs are explained the uses cases. This is the [third one](https://learn.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-client-creds-grant-flow#third-case-access-token-request-with-a-federated-credential) where the JWT assertion is created from a different identity provider rather than the proper application.

The thing is for GCP and MSFT it seems they refer to this auth flow as *workload identity federation* or *identity federation*; but I see nothing similar from AWS or nothing as standard in RFC; this is why not sure how to differentiate it from a "normal client credentials"